types/src/build_version/mod.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use async_std::sync::RwLock;
+use num_derive::FromPrimitive;
+use std::{
+    fmt::{Display, Formatter, Result as FmtResult},
+    process::Command,
+};
+
+use serde::Serialize;
+const BUILD_VERSION: &str = "0.10.2";
+
+//masks
+const MINOR_MASK: u32 = 0xffff00;
+const MAJOR_ONLY_MASK: u32 = 0xff0000;
+const MINOR_ONLY_MASK: u32 = 0x00ff00;
+const PATCH_ONLY_MASK: u32 = 0x0000ff;
+
+//api versions
+const FULL_API_VERSION: Version = new_version(0, 17, 0);
+const MINER_API_VERSION: Version = new_version(0, 15, 0);
+const WORKER_API_VERSION: Version = new_version(0, 15, 0);
+
+/// a constant for the block delay process in seconds
+pub const BLOCK_DELAY_SECS: u64 = 4;
+
+lazy_static! {
+    static ref CURRENT_COMMIT: String = {
+        let output = Command::new("git")
+            .args(&[
+                "describe",
+                "--always",
+                "--match=NeVeRmAtch",
+                "--dirty",
+                "2>/dev/null",
+            ])
+            .output()
+            .map(|s| s.stdout)
+            .unwrap_or_else(|_| {
+                Command::new("git")
+                    .args(&["rev-parse", "--short", "HEAD", "2>/dev/null"])
+                    .output()
+                    .map(|s| s.stdout)
+                    .unwrap_or_default()
+            });
+        String::from(std::str::from_utf8(&output).unwrap_or_default())
+    };
+    pub static ref BUILD_TYPE: RwLock<BuildType> = RwLock::new(BuildType::BuildDefault);
+    pub static ref RUNNING_NODE_TYPE: RwLock<NodeType> = RwLock::new(NodeType::Full);
```

Comment:
why rwlock? Always same?

Reply:
Would be injected depending

Reply:
The only point I was making here was that there is no need to wrap it in a mutex when it is never overwritten. Doesn't really matter but just no need for the overhead right now